nilearn/plotting/html_stat_map.py

```diff
@@ -0,0 +1,97 @@
+import os
+import tempfile
+import base64
+
+import numpy as np
+from nilearn import image, datasets
+
+from .._utils.extmath import fast_abs_percentile
+from .._utils.param_validation import check_threshold
+from .js_plotting_utils import get_html_template, HTMLDocument
+
+
+def _encode_nii(img):
+    fd, temp = tempfile.mkstemp(suffix='.nii')
```

Comment:
Isn't it possible to use StringIO / BytesIO (Python2 / 3) to avoid writing to a file?

Reply:
I had a quick look but couldn't find an obvious way to write a niimg to a
buffer. nibabel images' to_file_map really seem to want file objects; in
particulare they use same_file_as to check if the header and the image are in
the same file. Have you done this before?